examples/function-closure/function_closure.bal

```diff
@@ -0,0 +1,15 @@
+import ballerina/io;
+
+public function main() {
+    string[] names = ["Ana", "Alice", "Bob"];
+
+    // Define a function to modify and return the variable 'names' that are declared outside the scope.
+    var innerFunction = function() returns string[] {
+        // Access the variable `name` as closure within the `innerFunction`.
+        names.push("James");
+        return names;
+    };
+
+    io:println(innerFunction());
```

Comment:
```suggestion
    var addName= function(string value) returns string[] {
        // Access the variable `name` as closure within the `addName` inner function.
        names.push(value);
        return names;
    };

    io:println(addName("James"));
```

Intentionally picked `value` as the parameter instead of `name` to avoid confusion with the `names` variable.